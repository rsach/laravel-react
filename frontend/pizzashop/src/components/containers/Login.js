import { connect } from 'react-redux'
import { login} from '../../actions'


import Login from "../ui/Login";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onLogin(body){
            dispatch(
                login(body)
            )
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Login)
