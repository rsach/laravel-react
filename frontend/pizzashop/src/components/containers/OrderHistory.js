import { connect } from 'react-redux'


import OrderHistory from "../ui/OrderHistory";

const mapStateToProps = (state,props) =>
    ({
        orderHistory:state.orderHistory,
    });

const mapDispatchToProps = dispatch => ({});

export default connect(mapStateToProps,mapDispatchToProps)(OrderHistory)
