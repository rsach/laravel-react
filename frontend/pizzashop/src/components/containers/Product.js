import { connect } from 'react-redux'
import {addToCart, decreaseQuantity, increaseQuantity, removeFromCart} from '../../actions'
import Product from "../ui/Product";

const mapStateToProps = (state,props) =>
    ({
        product: props.product,
        cart: state.cart
    });

const mapDispatchToProps = dispatch =>
    ({
        onAddToCart(product){
            dispatch(
                addToCart(product)
            )
        },
        removeFromCart(product) {
            dispatch(
                removeFromCart(product)
            )
        },
        increaseQuantity(product) {
            dispatch(
                increaseQuantity(product)
            )
        },
        decreaseQuantity(product) {
            dispatch(
                decreaseQuantity(product)
            )
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Product)
