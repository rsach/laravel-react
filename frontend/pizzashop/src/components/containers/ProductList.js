import { connect } from 'react-redux'
import {addToCart, fetchProducts} from '../../actions'
import ProductList from "../ui/ProductList";

const mapStateToProps = (state,props) =>
    ({
        products:state.products.products,
    });

const mapDispatchToProps = dispatch =>
    ({
        onAddToCart(product){
            dispatch(
                addToCart(product)
            )
        },
        fetchProducts() {
            dispatch(
                fetchProducts('')
            )
        }
    });

export default connect(mapStateToProps,mapDispatchToProps)(ProductList)
