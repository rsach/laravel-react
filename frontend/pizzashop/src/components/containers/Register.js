import { connect } from 'react-redux'
import { register} from '../../actions'


import Register from "../ui/Register";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onSignUp(body){
            console.log(body)
            dispatch(
                register(body)
            )

            return body
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Register)
