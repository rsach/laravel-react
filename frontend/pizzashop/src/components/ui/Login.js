import React, {Component} from 'react';
import PropTypes from "prop-types";
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
// import AppBar from 'material-ui/AppBar';
import AppBar from '@material-ui/core/AppBar';



import history from "../../history";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";

class Login extends Component {
    constructor(props) {
        super(props);

        // reset login status


        this.state = {
            email: '',
            password: '',
            submitted: false
        };


        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }




    handleSubmit(e) {
        e.preventDefault();

        this.setState({ submitted: true });
        const { email, password } = this.state;
        const {onLogin=f=>f} = this.props;

        if (email && password) {

            onLogin({email, password});
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/register')
    }

    render() {


        return (

            <div>
                <MuiThemeProvider>
                    <div>
                        <AppBar
                            title="Login"
                        />
                        <div className="flex  flex-center">
                            <div className="form flex flex-column flex-center">


                                <TextField
                                    id="email"
                                    data-testid="email"
                                    variant="outlined"
                                    placeholder="Enter your Email"
                                    label="Email"
                                    onChange = {(event) => this.setState({email:event.target.value})}
                                />
                                <br/>
                                <TextField
                                    data-testid="password"

                                    id="password"
                                    variant="outlined"
                                    type="password"
                                    placeholder="Enter your Password"
                                    label="Password"
                                    onChange = {(event) => this.setState({password:event.target.value})}
                                />
                                <br/>
                                <div className="flex button-margin flex-space-between">
                                    <Button                    id="login"                 data-testid="login"
                                                                                variant="contained" color="primary"  onClick={(event) => this.handleSubmit(event)}>Login</Button>

                                        <Button id="register" variant="contained"        data-testid="register"
                                                onClick={(e) => this.handleRedirect(e)}  color="secondary"    >
                                            Register
                                        </Button>




                                </div>
                            </div>

                        </div>

                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
}


Login.propTypes = {
    onLogin: PropTypes.func.isRequired
};

export default Login

