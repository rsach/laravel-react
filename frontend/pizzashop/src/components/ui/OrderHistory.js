
import '../../stylesheets/index.scss';


import React from 'react'
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import AppBar from "material-ui/AppBar";




import OrderHistoryItem from "./OrderHistoryItem";



const OrderHistory = ({orderHistory = []})  =>
         (
             <div>
                <MuiThemeProvider>
                    <div>
                        <AppBar className="fixed" title="My Orders"></AppBar>
                        <div className="product-list flex flex-column">

                            {
                               (orderHistory || []).map((item, i) => <OrderHistoryItem key={i} item={item}></OrderHistoryItem>)

                            }








                        </div>
                    </div>
                </MuiThemeProvider>
            </div>
         );




OrderHistory.propTypes = {
    orderHistory: (props) => (!Array.isArray(props.orderHistory)) ?
        new Error('Order History must be an array') :
        null

};

export default OrderHistory