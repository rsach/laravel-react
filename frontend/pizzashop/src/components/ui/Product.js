
import PropTypes from 'prop-types';
import '../../stylesheets/index.scss';
import React from 'react'



const Product = ({product,onAddToCart=f=>f,removeFromCart=f=>f,increaseQuantity=f=>f,decreaseQuantity=f=>f, cart = []}) => {



    const cartData = cart.filter(res => res.id === product.id);




    return (
        <div className="product">
            <div className="image-container">
                <img id="image" alt={(product || {}).name} className="image" src={(product || {}).picture} />
                <p id="price">{(product || {}).price}</p>

            </div>
            <div className="flex flex-space-between">
                <p id="name">{(product || {}).name}</p>
                {

                    (cartData || []).length > 0 ?
                        (<div id="added">

                            <input id="quantity" disabled value={cartData[0].quantity}/>
                            <button id="increase" onClick={() => increaseQuantity(cartData[0])}>+</button>
                            {
                                cartData[0].quantity> 1 ?
                                    (<button id="decrease" onClick={() => decreaseQuantity(cartData[0])}>-</button>):
                                    (<button id="removeFromCart" onClick={() => removeFromCart(product)}>-</button>)
                            }


                        </div>

                        )

                     : (
                         <button id="addToCart" onClick={() => onAddToCart(product)}>ADD TO CART</button>
                    )}
            </div>


        </div>
    )
};


Product.propTypes = {
    onAddToCart:PropTypes.func,
    removeFromCart:PropTypes.func,
    increaseQuantity:PropTypes.func,
    decreaseQuantity:PropTypes.func,

    product: (props) => (!(typeof props.product === 'object')) ?
        new Error('Product must be an object') :
        null,


};

export default Product