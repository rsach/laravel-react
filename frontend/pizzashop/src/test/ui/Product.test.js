import React from 'react';
import { render } from '@testing-library/react';
import Login from "../../components/ui/Login";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Register from "../../components/ui/Register";
import ProductList from "../../components/ui/ProductList";
import PropTypes from "prop-types";
import Product from "../../components/ui/Product";
import OrderHistory from "../../components/ui/OrderHistory";

Enzyme.configure({ adapter: new Adapter() });


describe('<Product/> spec', () => {


    const propsFunc = {

        onAddToCart:jest.fn(),
        removeFromCart:jest.fn(),
        increaseQuantity:jest.fn(),
        decreaseQuantity:jest.fn(),
    };
    const product = {
        id: 1,
        name: 'Ila Kozey',
        price: 2,
        picture: 'https://lorempixel.com/640/480/?20457'

    };
    const cart = [
        {
            id:1,
            name:"Ila Kozey",
            price:2 ,
            picture:"https://lorempixel.com/640/480/?20457",
            created_at:"2019-12-29 14:27:07",
            updated_at:"2019-12-29 14:27:07",
            quantity:1
        }


    ];
    function setup() {
        const props = {
           ...propsFunc,

            product,
            cart: []
        };
        const wrapper = shallow(<Product {...props} />);
        return {
            props,  wrapper
        }
    }



    it('renders the component', () => {
        const {wrapper} = setup();
        expect(wrapper.firstChild).toMatchSnapshot();
    });


    it('add to cart', () => {
        const {wrapper, props} = setup();
        expect(wrapper.find('#addToCart').length).toBe(1);

        wrapper.find('#addToCart').simulate('click', props.product);
        expect(props.onAddToCart).toHaveBeenCalledTimes(1);

    });

    it('increase quantity ', () => {
        const props = { ...propsFunc, product, cart};
        const wrapper = shallow(<Product {...props} />);

        wrapper.find('#increase').simulate('click', props.cart[0]);
        expect(props.increaseQuantity).toHaveBeenCalledTimes(1);
    });

    it('decrease quantity', () => {
        const decCart = [{...cart[0]}];
        decCart[0].quantity = 2;
        const props = {...propsFunc, product, cart: decCart};
        const wrapper = shallow(<Product {...props} />);

        wrapper.find('#decrease').simulate('click', props.cart[0]);
        expect(props.decreaseQuantity).toHaveBeenCalledTimes(1);
        expect(wrapper.find('#added').length).toBe(1);



    });


    it('remove from cart', () => {
        const props = { ...propsFunc, product, cart};
        const wrapper = shallow(<Product {...props} />);

        wrapper.find('#removeFromCart').simulate('click', props.cart[0]);
        expect(props.removeFromCart).toHaveBeenCalledTimes(1);



    });

    it('renders error component', () => {


        const spy = jest.spyOn(console, 'error');

        const wrapper = shallow(<Product  />);
        expect(spy).toHaveBeenCalledTimes(1);

    });




});