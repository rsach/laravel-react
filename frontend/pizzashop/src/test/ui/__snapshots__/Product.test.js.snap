// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> spec renders the component 1`] = `undefined`;
